[Feature] GitHub webhook initialization

Sy1v4in opened the pull request:
This is the initialization of a webapp which can be plugged as a Github webhook through the `/github/webhook` route to log the posted informations.

Sy1v4in committed:
:tada: application initial commit

Sy1v4in committed:
:whale: add a Dockerfile

Sy1v4in committed:
:pencil: add heroku documentation

This document explains how to deploy the `releasy`
application on heroku and how to check the logs.

rehia commented:
I would recommend to use the template 
`<task|feat|docs|refactor|test|chore|perf|style>(<topic|context>): <message>`

rehia commented on build.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
What about making a `Makefile` ? :thinking:

Sy1v4in replied:
First of all, I would say that these files are generated by the dotnet template; I didn't write it 😉 

Then, why not using a Makefile, right?
To me the main advantage to use a Makefile as:
```
target: dependencies
    command
```
is that the target will be built incrementally when its dependencies have changed.
But, my knowledges of the `dotnet` ecosystem is too poor to write a Makefile: I don't really know what produces and what are the dependencies of a `dotnet restore` and a `dotnet build` commands 🤷‍♂️
But don't hesitate to make a concret proposition 😉

rehia replied:
From what I see, we need 2 targets
I'd suggest:

```suggestion
restore: ## Restore project dependencies
    dotnet restore src/Releasy

build: restore ## Build the whole project
    dotnet build src/Releasy

help:
    @grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'

.DEFAULT_GOAL := help
.PHONY: restore build help
```

benjamingarcia replied:
If you need build, can I test gradle on it?

Sy1v4in replied:
Of course, everybody can test what he or she wants 😉 And if the experiment is conclusive, why not to integrate it.
But, for now, I'm not sure I want to depend upon a new external tool. Moreover, I would like to be close enough to what is used in the Dockerfile.
And don't forget that, at this stage of the project, this build command run in about 5 sec. I'm not sure in the main painful point 💁‍♂️

Sy1v4in replied:
Fix in 851ae2c746bf54a2af4425bf8eb4450e656b6e84

rehia reviewed (commented):
Let me know abut the `Makefile`

Sy1v4in commented:
For the commit messages, I agree. I would just want to replace the commit types by their emoji equivalents described as [gitmoji](https://gitmoji.carloscuesta.me/).
It is more fun don't you think? 😜

benjamingarcia commented:
> I would recommend to use the template
> `<task|feat|docs|refactor|test|chore|perf|style>(<topic|context>): <message>`

it exist another list with `ci|build`instead of `chore`

Sy1v4in commented:
You're right @benjamingarcia we can improve the fist list provided by @rehia 
But, there are more than 60 gitmoji. So I think we can find our heart's desire 😄😍

Sy1v4in committed:
:recycle: (build) replace specific build files by a Makefile

Sy1v4in committed:
:rotating_light: (chore) remove Microsoft.AspNetCore.App version reference

Most usages of <PackageReference> require setting the Version attribute to specify the NuGet package version to be used. When using .NET Core 2.1 or 2.2 and referencing Microsoft.AspNetCore.App or Microsoft.AspNetCore.All, however, the attribute is unnecessary. The .NET Core SDK can automatically select the version of these packages that should be used.

Documentation recommendation: When referencing the Microsoft.AspNetCore.App or Microsoft.AspNetCore.All packages, do not specify their version.

asmodehn committed:
:see_no_evil: adding .idea to gitignore for Jetbrains Rider

rehia reviewed (commented):
This feature looks totally awesome !!!
